config/options.go

```diff
@@ -64,8 +66,18 @@ func Reporter(reporter jaeger.Reporter) Option {
 
 // Observer can be registered with the Tracer to receive notifications about new Spans.
 func Observer(observer jaeger.Observer) Option {
+	old := jaeger.OldObserver{Obs: observer}
```

Comment:
I went with the 1st option, since, the 2nd option will still make old observer public.